src/phpDebug.ts

```diff
@@ -437,7 +441,9 @@ class PhpDebugSession extends vscode.DebugSession {
     ) {
         try {
             const fileUri = convertClientPathToDebugger(args.source.path!, this._args.pathMappings)
-            const connections = Array.from(this._connections.values())
+            // If there is a waiting connection, only use use that as it is likely being called to init
+            // the breakpoints on the new connection
```

Comment:
What concerns me about this fix is the word "likely". "likely" to me means "race condition". Can this be fixed without having to rely on the likelihood of something?

I think fundamentally the problem here is that for a single _VS Code <-> debug adapter_ connection we need to persist the breakpoints to _multiple_ connected PHP processes. Some may block because they are mid-execution, some may succeed, some may fail, but we only have a single request from VS Code to handle and respond to (with verified or error).

@weinand any advice how to handle this scenario best?

Reply:
@felixfbecker Yeah, a more stable fix would be to somehow account for success, failed, or blocked and the blocked would not stop `configurationDone` from firing.  Ideally if there was some concrete way to tell that the breakpoint request was part of the process of initializing a new connection, I think it is still a good idea to only run it on the new connection in that case.

The tricky part here is that as far as I can tell these are initiated from vscode itself.  I used the word `likely` because of that, since _pendingConnections is not empty that means a new connection was just added.  For it to "not" be because of the new connection, the user would have had to do something like change a breakpoint at the precise moment a new connection was established, which is always a possibility.

Maybe we first: when doing something that executes code, set a flag like `_isLocked` that gets un-set when the response resolves.  Then when making the request to do something with breakpoints, if `connection._isLocked` it would queue the command but if there are more than one connection, as long as one succeeds it would continue.  That would allow the after configuration stuff to run for the new connection.

I hope that makes sense?

Reply:
In other words, instead of `Promise.All`, use something like `Promise.Any`, is that even a thing?